lib/services/monitoring.rb

```diff
@@ -17,25 +18,53 @@ def self.set_repo_latest_release(repo, client)
         return
       end
 
-      if releases.nil? || releases.empty?
-        begin
-          tags = client.tags(repo_path)
-        rescue Octokit::Error => e
-          log_error e
-          return
-        end
+      begin
+        tags = client.tags(repo_path)
+      rescue Octokit::Error => e
+        log_error e
+        return
+      end
+
+      if tags.nil? || tags.empty?
+        return
+      end
+
+      latest_tag_index = tags.find_index { |tag| important_tag? tag.name }
+      latest_tag = tags[latest_tag_index]
+      begin
+        commit = client.commit(repo_path, latest_tag.commit.sha)
+      rescue Octokit::Error => e
+        log_error e
+        return
+      end
+      latest_tag_date = commit.commit.author.date
+      latest_tag = latest_tag.name
+
+      if releases.nil? || releases.empty? || latest_tag_date - releases.first.published_at > 30
+        latest_release_date = commit.commit.author.date
 
-        if tags.nil? || tags.empty?
-          return
-        end
-        latest_tag = tags.first.name
         begin
-          commit = client.commit(repo_path, tags.first.commit.sha)
+          changelog_file = client.contents(repo_path)
+                            .filter { |file| file.name.match(Regexp.new("changelog.md", Regexp::IGNORECASE)) }
+                            .first
+          unless changelog_file.nil?
```

Comment:
There are a lot of nested `unless`/`if` statements and it can make it tough to read the code. I'd recommend writing it in a way you can do the check once, or flatten out the conditionals, e.g.,

```
changelog_file = ...
download = changelog_file&.download_url&.then { URI.open(_1) }
changelog = download&.read
return if changelog.nil?

...
```